app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,70 @@
+class TweetsController < ApplicationController
+
+
+    get '/' do
+        erb :index
+    end
+    
+    get '/tweets' do
+        if logged_in?
+            erb :'tweets/tweets'   
+        else
+            redirect '/login'
+        end
+    end
+    
+    get '/tweets/new' do
+        if logged_in?
+            erb :"tweets/create_tweet"
+        else
+            redirect '/login'
+        end
+    end
+    
+    get '/tweets/:id' do
+        if logged_in?
+            @tweet = Tweet.find_by_id(params["id"])
+            erb :"tweets/show_tweet"
+        else
+            redirect "/login"
+        end
+    end
+    
+    get '/tweets/:id/edit' do
+        if logged_in?
+            @tweet = Tweet.find_by_id(params["id"])
+            erb :'tweets/edit_tweet'
+        else
+            redirect '/login'
+        end
+    end
+
+    patch '/tweets/:id' do
+        @tweet = Tweet.find_by_id(params["id"])
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...